public/tools/clientSetup4RMT.sh

```diff
@@ -0,0 +1,306 @@
+    cat << EOT >&2
+
+  Usage: $0 <registration URL> [--regcert <url>] [--namespace <namespace>] [--regdata <filename>] [--de-register]
+  Usage: $0 --host <hostname of the RMT server> [--regcert <url>] [--namespace <namespace>] [--regdata <filename>] [--de-register]
+  Usage: $0 --host <hostname of the RMT server> [--fingerprint <fingerprint of server cert>] [--yes] [--regdata <filename>] [--de-register]
+         configures a SLE client to register against a different registration server
+
+  Example: $0 https://rmt.example.com/
+  Example: $0 --host rmt.example.com --namespace web
+  Example: $0 --host rmt.example.com --regcert http://rmt.example.com/certs/rmt.crt
+
+  If --namespace is omitted, no namespace is set and this results in using the
+  default production repositories.
+EOT
+
+exit 1
+}
+
+AUTOACCEPT=""
+FINGERPRINT=""
+REGDATA=""
+REGURL=""
+VARIABLE=""
+NAMESPACE=""
+DE_REGISTER=""
+while true ; do
+    case "$1" in
+        --fingerprint) VARIABLE=FINGERPRINT;;
+        --host) VARIABLE=S_HOSTNAME;;
+        --regcert) VARIABLE=REGCERT;;
+        --regdata) VARIABLE=REGDATA;;
+        --namespace) VARIABLE=NAMESPACE;;
+        --de-register) DE_REGISTER="Y";;
+        --yes) AUTOACCEPT="Y";;
+        "") break ;;
+        -h|--help) usage;;
+        https*) REGURL=$1;;
+        *) usage "Unknown option $1";;
+    esac
+    if [ -n "$VARIABLE" ] ; then
+        test -z "$2" && usage "Option $1 needs an argument"
+        eval $VARIABLE=\$2
+        shift
+        VARIABLE=""
+    fi
+    shift
+done
+
+if [ `id -u` != 0 ]; then
+    echo "You must be root. Abort."
+    exit 1;
+fi
+
+if [ -n "$S_HOSTNAME" -a -e $SUSECONNECT ]; then
+    REGURL="https://$S_HOSTNAME/"
+fi
+
+if [ -z "$REGURL" ]; then
+    echo "Missing registration URL. Abort."
+    usage
+fi
+
+if ! echo $REGURL | grep "^https" > /dev/null ; then
+    echo "The registration URL must be a HTTPS URL. Abort."
+    exit 1
+fi
+
+if ! echo $NAMESPACE | grep -E "^[a-zA-Z0-9_-]*$" > /dev/null ; then
+    echo "Invalid characters in namespace. Allowed are [a-zA-Z0-9_-]. Abort."
+    exit 1
+fi
+
+# BNC #516495: Changing supportconfig URL for uploading tarbals
+if [ "${S_HOSTNAME}" != "" ]; then
+    if [ -e "${SUPPORTCONFIG}" ]; then
+        S_ENTRY="http://${S_HOSTNAME}/upload?appname=supportconfig\&file={tarball}"
+
+        ${SED} --in-place "s|${SUPPORTCONFIGENTRY}[ \t]*=.*$|${SUPPORTCONFIGENTRY}='${S_ENTRY}'|" ${SUPPORTCONFIG}
+    fi
+fi
+
+if [ -z "$REGCERT" ]; then
+    CERTURL=`echo "$REGURL" | awk -F/ '{print "https://" $3 "/rmt.crt"}'`
+else
+    CERTURL="$REGCERT"
+fi
+
+if [ "$AUTOACCEPT" = "Y" ] && [ -z "$FINGERPRINT" ]; then
+    echo "Must specify fingerprint with auto accept and auto registration. Abort."
+    exit 1
+fi
+
+if [ ! -z "$REGDATA" ] && [ ! -f "$REGDATA" ]; then
+    echo "Specified file $REGDATA not found."
+    exit 1
+fi
+
+
+if [ ! -x $OPENSSL ]; then
+    echo "openssl command not found. Abort."
+    exit 1
+fi
+
+if [ ! -x $CP ]; then
```

Comment:
Are the checks for `cp` and `cat` really required? Since they are in `coreutils` and I can not imagine a generic system without this package

Reply:
It's a bit too paranoid for my taste, but this script is borrowed from SMT package with minimal changes (older SLE11 openssl stuff and `suse_register` references removed) -- I don't see a reason to change this.
We are considering either moving this script to SUSEConnect package or implementing a feature in SUSEConnect itself to allow for importing certificates.

Reply:
Yes true. Ok than. But its somewhat funny `cp` is checked but `mktemp` is ok too use without checking even its also in `coreutils`. Anyway a connect implementation is preferable I guess.